### PR TITLE
Opendrive signals feature

### PR DESCRIPTION
**Pull request by ishaan95**
Signals class added to pyodrx module. Few changes to the Roads class and to the create_straight_roads() function to accomodate testing this feature.

### CODE REVIEW

File: pyodrx/generators.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def create_road(geometry,id,left_lanes = 1, right_lanes = 1,road_type=-1,center_
 
     return Road(id,pv,lanes,road_type=road_type)
 
-def create_straight_road(road_id,length=100,junction = -1, n_lanes=1, lane_offset=3):
+def create_straight_road(road_id, signals=None, length=100,junction = -1, n_lanes=1, lane_offset=3):
```
Comment (mander76) [inline pyodrx/generators.py:98]: Please add to documentation aswell

**Review by mander76 — CHANGES_REQUESTED** [review]
Very nice, thanks for contributing!

I have added some comments in the commit (mostly minor things, for documentation)

Main thing to change, add a add_signal to the Signals class instead of using append.

Could you create one example in the examples folder aswell?

**Comment by jakobkaths** [comment]
Hi, as I mentioned on the related issue before, I planned to introduce the "object" as well and since they're somewhat related, I would introduce a base class that both signals and objects can use. What would be your preferred way to do this, @ishaan95 and @mander76 ? Should I wait until this pull request is merged and then introduce the changes (will also touch some of the things that have been introduced here) or should I base it on this and already create a pull request? @ishaan95 do you know when you will address the changes requested here?

File: pyodrx/opendrive.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ class Road():
                 write a open scenario xml
                 
     """
-    def __init__(self,road_id,planview,lanes, road_type = -1,name=None, rule=None):
+    def __init__(self,road_id,planview,lanes, road_type = -1,name=None, rule=None, signals=None):
```
Comment (jakobkaths) [inline pyodrx/opendrive.py:112]: What was the reason to integrate this here? I would have expected an add_signal method similar to the roadmarks for lanes. Is there a preferred way for integration?

Comment (mander76) [reply pyodrx/opendrive.py:112]: I usually think that if you can add multiple entries of one class you do a add_ functionality, like roadmarks, or lane_sections to lanes. 
Here you either add signals (which is a single entry) or you add Signal directly to the road (then create the signals in road).
So both "work" in this case, following how the rest is built up.

**Comment by mander76** [comment]
@jakobkaths Sounds good to introduce the object, and if they have similarities it's not bad to combine. Then I guess this will change this commit?
Would be preferable to wait, @ishaan95 if it doesn't take too long to do this review?

**Comment by ishaan95** [comment]
I'll add the changes to the review today. I'll have a look at the base object suggested by @jakobkaths. I think it would be better to wait until my changes have been merged and then we can make further changes.

File: pyodrx/signals.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                   "zOffset": str(self.zOffset), "country": str(self.country), "type": str(self.type),
+                   "subtype": str(self.subtype), "value": str(self.value)}
+        return retdict
+
+    def get_element(self):
+        element = ET.Element('signal', attrib=self.get_attributes())
+        return element
+
+
+class Dependency:
```
[the remark this answers is not in the document]
Comment (ishaan95) [reply pyodrx/signals.py:114]: Yes this is for future implementations according to the Opendrive 1.6 documentation.

**Pull request merged**